includes/class-newspack-ads-settings.php

```diff
@@ -0,0 +1,374 @@
+<?php
+/**
+ * Newspack Ads Settings
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack Ads Settings Class.
+ */
+class Newspack_Ads_Settings {
+
+	const API_NAMESPACE      = 'newspack-ads/v1';
+	const API_CAPABILITY     = 'manage_options';
+	const OPTION_NAME_PREFIX = '_newspack_ads_';
+
+	/**
+	 * Initialize settings.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints needed to fetch and update settings.
+	 */
+	public static function register_api_endpoints() {
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings-list',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ __CLASS__, 'api_get_settings_list' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+			]
+		);
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ __CLASS__, 'api_get_settings' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+				'args'                => [
+					'section' => [
+						'default'           => '',
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ __CLASS__, 'api_update_section' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+				'args'                => [
+					'section'  => [
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+					'settings' => [
+						'required'          => true,
+						'sanitize_callback' => [ __CLASS__, 'api_sanitize_settings' ],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Check capabilities for using API.
+	 *
+	 * @return bool|WP_Error True or error object.
+	 */
+	public static function api_permissions_check() {
+		if ( ! current_user_can( self::API_CAPABILITY ) ) {
+			return new \WP_Error(
+				'newspack_ads_rest_forbidden',
+				esc_html__( 'You cannot use this resource.', 'newspack-ads' ),
+				[
+					'status' => 403,
+				]
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Sanitize settings coming from the API.
+	 *
+	 * @param array           $settings Settings to sanitize.
+	 * @param WP_REST_Request $request  Full details about the request.
+	 *
+	 * @return array Sanitized settings.
+	 */
+	public static function api_sanitize_settings( $settings, $request ) {
+		$section            = (string) $request['section'];
+		$sanitized_settings = [];
+		foreach ( $settings as $key => $value ) {
+			$config = self::get_setting_config( $section, $key );
+			settype( $value, $config['type'] );
+			$sanitized_settings[ $key ] = $value;
+		}
+		return $sanitized_settings;
+	}
+
+	/**
+	 * Update setting section.
```

Comment:
Description should be updated.

Reply:
See https://github.com/Automattic/newspack-ads/pull/220